src/rules/no-firestore-object-arrays.ts

```
@@ -81,68 +81,33 @@ const unwrapArrayElementType = (node: TSESTree.TypeNode): TSESTree.TypeNode => {
-      return false;
-  }
-};
-
-const isImmediatelyWrappedByArraySyntax = (node: TSESTree.Node): boolean => {
-  let parent = (node as TSESTree.Node).parent as TSESTree.Node | undefined;
-  // Skip non-semantic wrappers (readonly/parens)
-  while (parent) {
+// Determine whether this type node appears in a Firestore model type context
+// i.e., within an interface or type alias declaration, and not within variable/function annotations
+const isInModelTypeContext = (node: TSESTree.Node): boolean => {
+  let current: TSESTree.Node | undefined = (node as TSESTree.Node).parent as TSESTree.Node | undefined;
```

ESLint

[error] Replace `·TSESTree.Node` with `⏎····|·TSESTree.Node⏎···`

(prettier/prettier)